examples/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"os"
+	"reflect"
+
+	"github.com/digitalocean/firebolt/util"
+
+	"github.com/digitalocean/firebolt/examples/shared"
+
+	"github.com/digitalocean/firebolt/executor"
+	"github.com/digitalocean/firebolt/metrics"
+
+	"github.com/digitalocean/firebolt/examples/kafkatokafka"
+
+	"github.com/digitalocean/captainslog"
+
+	"github.com/digitalocean/firebolt/node"
+)
+
+func main() {
+	exampleName := os.Args[1]
+	util.WaitForPort(nil, 9092)
+
+	// run the selected example
+	if exampleName == "kafkatokafka" {
+		runKafkaToKafka()
+	} else {
+		panic("unknown example: " + exampleName)
+	}
+
+	// produce test data, consume any results
+	shared.ProduceTestData(10)
+	shared.ConsumeTestData("ktk-dest", 10)
+}
+
+func runKafkaToKafka() {
+	metrics.Init("kafkatokafka")
+
+	// register the parser node
+	node.GetRegistry().RegisterNodeType("parser", func() node.Node {
+		return &kafkatokafka.Parser{}
+	}, reflect.TypeOf([]byte(nil)), reflect.TypeOf(captainslog.SyslogMsg{}))
+
+	// register the jsonbuilder node
+	node.GetRegistry().RegisterNodeType("jsonbuilder", func() node.Node {
+		return &kafkatokafka.JSONBuilder{}
+	}, reflect.TypeOf(captainslog.SyslogMsg{}), reflect.TypeOf([]byte(nil)))
+
```

Comment:
Would love a bit of an explanation here about how these things connect together. I know I could go read the documentation, but it's always cool to me when examples have more or less everything I need to understand what the project does. Explaining how these names connect to the config and stuff, basically ELI5 the example in full. imo the comments on all this stuff can't be too verbose.

Above I see that we're producing/consuming test data, how does what we're setting up here connect with that? Like it's obvious to see that it's related somehow, but only looking at this, I can't really connect the dots without more explanation.

Reply:
thats great feedback thanks!